### PR TITLE
[16.0][ADD] account_analytic_document_date: New module account_analyt…

### PR DESCRIPTION
**Pull request by BernatObrador**
This module adds an other date to the account.move for analytics proposits

Usage:

1. **Set Document Date on Invoice:**
   - Navigate to the invoice and set the document date. 
   - If the document date is not set, it will automatically be assigned the invoice date upon confirmation.

2. **Analytic Line Generation:**
   - If any line has analytic information, the generated analytic entry will inherit the document date.

3. **Setting Document Date in Reconcile View:**
   - You can also set the document date from the reconciliation view, but this option is available only during manual reconciliation.

cc https://github.com/APSL 160768
@miquelalzanillas @lbarry-apsl @javierobcn @peluko00 @ppyczko please review

### CODE REVIEW

**Review by peluko00 — APPROVED** [review]
LGTM, code review and tested in runboat

**Review by ppyczko — APPROVED** [review]
LGTM, tested in runboat

**Comment by BernatObrador** [comment]
@pedrobaeza Could you review please?

File: account_analytic_document_date/i18n/ca_ES.po
Comment (pedrobaeza) [inline account_analytic_document_date/i18n/ca_ES.po:1]: It should be `ca.po`

**Review by pedrobaeza — COMMENTED** [review]
I don't understand the goal of having another date. I would understand that you want a different date for the analytic line, but not having dual dates.

**Comment by BernatObrador** [comment]
@pedrobaeza 
The module introduces a new field in the accounting records to specify the "Accrual Date" for income and expenses. This field is crucial for accurately tracking when the revenue or expense is earned or incurred, regardless of the invoice creation date.

For instance, if an invoice is created in December but refers to sales made in November, the Accrual Date allows us to reflect this correctly in our accounting records. This ensures compliance with accrual-based accounting principles and provides better insights into the financial period to which transactions belong.

This feature is particularly valuable for businesses that need to differentiate between the accounting period of transaction occurrence and the timing of document issuance.

File: account_analytic_document_date/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+{
+    "name": "Account Analytic Document Date",
+    "summary": "Account Analytic Document Date",
+    "version": "16.0.1.0.0",
+    "website": "https://github.com/OCA/account-analytic",
+    "author": (
+        "(Nagarro - APSL), Miquel Pascual, Bernat Obrador,"
+        "Odoo Community Association (OCA)"
+    ),
+    "maintainers": ["mpascual@apsl.net, bobrador@apsl.net"],
```
Comment (pedrobaeza) [inline account_analytic_document_date/__manifest__.py]: These are Github logins, not emaisl.

File: account_analytic_document_date/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2024 (APSL - Nagarro) Miquel Pascual, Bernat Obrador
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Account Analytic Document Date",
+    "summary": "Account Analytic Document Date",
+    "version": "16.0.1.0.0",
+    "website": "https://github.com/OCA/account-analytic",
+    "author": (
+        "(Nagarro - APSL), Miquel Pascual, Bernat Obrador,"
```
Comment (pedrobaeza) [inline account_analytic_document_date/__manifest__.py]: This is the company, not the persons

File: account_analytic_document_date/static/description/icon.svg
Comment (pedrobaeza) [inline account_analytic_document_date/static/description/icon.svg:1]: Don't put this if it's not modified from the standard one (which is the case).

File: account_analytic_document_date/models/account_move.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2024 (APSL - Nagarro) Miquel Pascual, Bernat Obrador
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    document_date = fields.Date()
```
Comment (pedrobaeza) [inline account_analytic_document_date/models/account_move.py]: If it's a date only for analytic, it shouldn't be called `document_date`, but `analytic_document_date` or `analytic_date`.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: account_analytic_document_date/models/account_move.py (hunk 3 above)
Comment (BernatObrador) [reply account_analytic_document_date/models/account_move.py]: Indeed, it is used for analytic purposes, but it is also visible in the account module, so it serves both purposes. For this reason, I believe the name 'document_date' remains appropriate.

Comment (pedrobaeza) [reply account_analytic_document_date/models/account_move.py]: But it creates confusion on users thinking that date is going to be used for example for financial reports. It should be renamed to `analytic_document_date` IMO.

**Comment by BernatObrador** [comment]
@pedrobaeza Could you please review again?

File: account_analytic_document_date/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2024 (APSL - Nagarro) Miquel Pascual, Bernat Obrador
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Account Analytic Document Date",
+    "summary": "Account Analytic Document Date",
+    "version": "16.0.1.0.0",
+    "website": "https://github.com/OCA/account-analytic",
+    "author": ("(Nagarro - APSL), Odoo Community Association (OCA)"),
```
Comment (pedrobaeza) [inline account_analytic_document_date/__manifest__.py]: Is this the usual form (between parenthesis) of your company name?

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**